atlp-rwanda/bravo-bn-be
# Feature#182201735 travel request updates notifications

#### What does this PR do?
> - Email and InApp notifications on travel request updates
#### Description of Task to be completed?
> - integrate socket.io
> - emit events on request updates
> - create in-app notifications on travel request updates
> - Send Email Notifications
#### How should this be manually tested?
> - clone this repository
> - checkout to this branch `ft-trip-updates-notifications-182201735`
> - Set `NODE_ENV=production` to send email notifications
> - run `npm install` to install dependencies
> - run migrations using `npm run migrate`
> - run `npm run start:prod` to start the production server
> - Access the static page on `localhost:4000 or your port nber` 
> - Login and paste the token in the input on the page and click on connect button
 to create a socket connection
> - Send a request like create trip request on the api
> - Go back on the static page and you will see notifications already in the console
#### What are the relevant pivotal tracker stories?
> - https://www.pivotaltracker.com/n/projects/2569105/stories/182201735
#### Screenshots(If appropriate)
> <img width="1240" alt="Screen Shot 2022-07-23 at 01 19 58" src="https://user-images.githubusercontent.com/73250459/180598205-2a123d93-1149-43db-832a-79577e26b72b.png">

#### Email Notification
> <img width="1240" alt="Screen Shot 2022-07-23 at 01 20 29" src="https://user-images.githubusercontent.com/73250459/180599583-58c7fd01-082c-4403-96eb-2692db52bbee.png">

#### In-App Notification
> <img width="1240" alt="Screen Shot 2022-07-23 at 01 20 11" src="https://user-images.githubusercontent.com/73250459/180598266-82981069-b7f4-434c-abf3-37c118ada047.png">
